### PR TITLE
Features/sdk announce stream 178308136

### PR DESCRIPTION
**Pull request by aneyzberg**
Problem
=======
We need a way to be able to consistently listen for and retrieve batchAnnounce events that happen on chain
[link to Pivotal Tracker #178308136](https://www.pivotaltracker.com/story/show/178308136)

Solution
========
We added a function that sets up a listener to listen for and returns batchAnnounce events from the chain. 
This function takes a callback and a filter - and will filter through events and then invoke the callback for each filtered event.

with @wilwade 

Change summary:
---------------
* Add a new `subscription.ts` module
* Add functionality to set up listener in function `batchAnnounceEvents`
* Add functionality to filter events in `batchAnnounceEvents`
* have `batchAnnounceEvents deal with retrieving both bast and current events`
* Add spec
* Add `utilities.ts` file in test with a `checkNumberOfFunctionCalls` function that waits for our callback to get called
* use new ganache container in circle so we no longer need to look for the contracts and deploy them to a container.

### CODE REVIEW

File: .github/workflows/main.yml
Comment (wilwade) [inline .github/workflows/main.yml:12]: 🎉

File: src/core/contracts/announcement.ts
Comment (wilwade) [inline src/core/contracts/announcement.ts:6]: comment

File: src/core/contracts/subscription.test.ts
Comment (wilwade) [inline src/core/contracts/subscription.test.ts:16]: reset does everything clear does (+ extra)
```suggestion
    jest.resetAllMocks();
```

Actually, you don't reuse any mocks, so:

```suggestion
```

File: src/core/contracts/subscription.test.ts
Comment (wilwade) [inline src/core/contracts/subscription.test.ts:22]:
```suggestion
```

File: src/core/utilities/index.ts
Comment (wilwade) [inline src/core/utilities/index.ts:8]: Not sure why these changed.

File: src/core/utilities/index.ts
Comment (wilwade) [inline src/core/utilities/index.ts:17]: Same

**Review by wilwade — DISMISSED** [review]
Few small things

File: src/core/contracts/subscription.test.ts
Comment (aneyzberg) [reply src/core/contracts/subscription.test.ts:16]: oh yes. This I meant to clean up a while ago.

File: src/core/utilities/index.ts
Comment (aneyzberg) [reply src/core/utilities/index.ts:8]: I am not sure either.

**Review by wilwade — COMMENTED** [review]
lgtm. We paired some on it, so be nice to get another set of eyes on it before merging.

File: src/core/contracts/subscription.ts
Comment (rosalinekarr) [inline src/core/contracts/subscription.ts:40]: Can we rename this to something like `subscribeToBatchAnnounceEvents` or just `subscribe`? This naming makes it seem like this method is for announcing a batch.

**Review by rosalinekarr — APPROVED** [review]
LGTM, only non-blocking feedback

File: src/core/contracts/subscription.ts
Comment (rosalinekarr) [reply src/core/contracts/subscription.ts:40]: Also, depending on whether "batch" is a verb or a noun here, this may be breaking our function naming convention.

Comment (aneyzberg) [reply src/core/contracts/subscription.ts:40]: It is in the subscription module. But I do see your point. I can do that.

**Pull request merged**
